### PR TITLE
feat(data-security): data masking support columns in view

### LINKED ISSUE #46: [Feature]: Data masking supports columns in database view
### Describe your use case

ODC provide data masking for database tables since v4.2.0, but the database views are not taken into consideration.
In the current situation, sensitive data can be access  illegally through creating a view and select data from this view.

1. So, I propose to incorporate views into data desensitization control.

### Describe the solution you'd like

1. Allow setting columns in views as sensitive columns；
2. When extracting columns from SQL, just make views into consideration.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-feature
module-data security

#### What this PR does / why we need it:
1. Extend db-browser to listTableColumns and listViewColumns
2. Make data-masking support masking columns in view

#### Which issue(s) this PR fixes:
Close #46 

#### Special notes for your reviewer:
This PR is WIP, merge will be enbaled after self-testing.

### CODE REVIEW

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLBetween2277And3XSchemaAccessor.java [hunk 1]
```diff
@@ -231,4 +233,54 @@ public List<DBTableColumn> listTableColumns(String schemaName, String tableName)
...
+
+    @Override
+    public Map<String, List<DBTableColumn>> listBasicViewColumns(String schemaName) {
+        MySQLSqlBuilder sb = new MySQLSqlBuilder();
+        sb.append("select table_name from information_schema.views where table_schema=");
+        sb.value(schemaName);
+        List<String> viewNames = jdbcOperations.query(sb.toString(), (rs, rowNum) -> rs.getString("table_name"));
+        List<DBTableColumn> columns = new ArrayList<>();
+        for (String viewName : viewNames) {
+            columns.addAll(fillViewColumnInfoByDesc(schemaName, viewName));
```
Comment (yizhouxw) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLBetween2277And3XSchemaAccessor.java:253]: performace risk , avoid dict view query in looop

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleLessThan400SchemaAccessor.java [hunk 2]
```diff
@@ -18,17 +18,14 @@
...
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import org.springframework.jdbc.core.BeanPropertyRowMapper;
 import org.springframework.jdbc.core.JdbcOperations;
 
-import com.oceanbase.tools.dbbrowser.model.DBObjectIdentity;
-import com.oceanbase.tools.dbbrowser.model.DBSynonymType;
+import com.oceanbase.tools.dbbrowser.model.*;
```
Comment (yizhouxw) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleLessThan400SchemaAccessor.java:27]: avoid import `*`

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.oceanbase.tools.dbbrowser.model.DBTableColumn;
+
+import lombok.Data;
+
+/**
+ * @author gaoda.xy
+ * @date 2023/9/7 14:57
+ */
+@Data
+public class ListColumnsResp {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java:31]: seems we can remove ListColumnsResp class, how about use `ListResponse<DatabaseColumn>` as response in Controller  ?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    private List<DatabaseColumn> databaseColumns;
+
+    public static ListColumnsResp of(List<DatabaseColumn> databaseColumns) {
+        ListColumnsResp resp = new ListColumnsResp();
+        resp.setDatabaseColumns(databaseColumns);
+        return resp;
+    }
+
+    @Data
+    public static class DatabaseColumn {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java:45]: class name: DatabaseColumn -->  DatabaseWithAllColumns ?
field names:
- tableColumns --> table2Columns
- viewColumns --> view2Columns

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/SensitiveColumnController.java [hunk 5]
```diff
@@ -55,6 +56,13 @@ public class SensitiveColumnController {
     @Autowired
     private SensitiveColumnService service;
 
+    @ApiOperation(value = "listColumns", notes = "List table columns and view columns")
+    @RequestMapping(value = "/listColumns", method = RequestMethod.GET)
```
Comment (yizhouxw) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/SensitiveColumnController.java:60]: `@RequestMapping(value = "/", method = RequestMethod.GET)`
 
return `ListResponse<SensitiveColumn>`

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLBetween2277And3XSchemaAccessor.java (hunk 1 above)
Comment (smallsheeeep) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLBetween2277And3XSchemaAccessor.java:253]: This is a work around way, more details:
image

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleLessThan400SchemaAccessor.java (hunk 2 above)
Comment (smallsheeeep) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleLessThan400SchemaAccessor.java:27]: Done! Blame to IDEA.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java (hunk 3 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java:31]: Done!

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java (hunk 4 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/datasecurity/model/ListColumnsResp.java:45]: Okay!

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/SensitiveColumnController.java (hunk 5 above)
Comment (smallsheeeep) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/SensitiveColumnController.java:60]: `@RequestMapping(value = "/", method = RequestMethod.GET)` should used to list sensitive columns and is occupied yet.

**Review by krihy — APPROVED** [review]
LGTM

**Pull request merged**